src/python/WMCore/Services/WMStats/WMStatsReader.py

```diff
@@ -370,26 +378,36 @@ def isWorkflowCompletedWithLogCollectAndCleanUp(self, requestName):
         reqInfoInstance = RequestInfo(requestInfo[requestName])
         return reqInfoInstance.isWorkflowFinished()
 
-    def getTaskJobSummaryByRequest(self, requestName, sampleSize=1):
-
+    def getTaskJobSummaryByRequest(self, requestName, sampleSize=1, usePycurl=True):
+        reqStart = time.time()
         options = {'reduce': True, 'group_level': 5, 'startkey': [requestName],
                    'endkey': [requestName, {}]}
         results = self._getCouchView("jobsByStatusWorkflow", options)
+
         jobDetails = {}
-        for row in results['rows']:
-            # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
-            startKey = row["key"][:4]
-            endKey = []
-            site = row["key"][4]
-            if site:
-                startKey.append(site)
-
-            endKey.extend(startKey)
-            endKey.append({})
-            numOfError = row["value"]
-
-            jobInfo = self.jobDetailByTasks(startKey, endKey, numOfError, sampleSize)
+        if usePycurl is True:
+            jobInfo = getTaskJobSummaryByRequestPycurl(results, sampleSize, self.serviceOpts)
             jobDetails = nestedDictUpdate(jobDetails, jobInfo)
+        else:
+            # then it is sequential
+            for row in results['rows']:
+                # row["key"] = ['workflow', 'task', 'jobstatus', 'exitCode', 'site']
+                startKey = row["key"][:4]
+                endKey = []
+                site = row["key"][4]
+                if site:
+                    startKey.append(site)
+
+                endKey.extend(startKey)
+                endKey.append({})
+                numOfError = row["value"]
+
+                jobInfo = self.jobDetailByTasks(startKey, endKey, numOfError, sampleSize)
+                jobDetails = nestedDictUpdate(jobDetails, jobInfo)
+        callRuntime = round((time.time() - reqStart), 3)
+        msg = f"Retrieved job details (pycurl mode: {usePycurl}) for {requestName} in "
+        msg += f"{callRuntime} seconds, with a total of {len(results['rows']) + 1} CouchDB calls"
+        print(msg)
```

Comment:
Note that I put this print statement here because it's the only way to get this logged into the service access_log. If we prefer, I can remove it. But I think it's beneficial to keep it around for a while, at least.